### ISSUE TITLE
torch.fake_quantize_per_tensor_affine will affect the results of the model if the input memory is not contiguous

### ISSUE THREAD

**Issue by aksenventwo** [pytorch/pytorch#75009]
### 🐛 Describe the bug

When I'm quantizing shufflenet in torchvision, `torch.fake_quantize_per_tensor_affine` will affect the results of the model if the input memory is not contiguous. Affected also include `torch._fake_quantize_learnable_per_tensor_affine` api. 

I am using version 1.8.1+cu11, I guess the latest version should also have this problem.

[Shufflenetv2](https://github.com/pytorch/vision/blob/93104c1697384273a75381d57be2f872265eeefb/torchvision/models/shufflenetv2.py#L95)

```python

    def forward(self, x: Tensor) -> Tensor:
        if self.stride == 1:
            x1, x2 = x.chunk(2, dim=1)
            out = torch.cat((x1, self.branch2(x2)), dim=1)
        else:
            out = torch.cat((self.branch1(x), self.branch2(x)), dim=1)

        out = channel_shuffle(out, 2)

        return out
```

When I use `torch.fake_quantize_per_tensor_affine` for `x2` it gives completely wrong result, But if I do contiguous operation on `x2` , the result is correct.

To get around this problem, I recommend always adding the `contiguous` action 

https://github.com/pytorch/pytorch/blob/ce700da7f4b3f2930a83f2f8995c0039c7838faf/torch/ao/quantization/fake_quantize.py#L180

and 

https://github.com/pytorch/pytorch/blob/8c05f44fe2bcb6b13758addc5a8ff9826e3a2601/torch/ao/quantization/_learnable_fake_quantize.py#L150

It would be better if this could be solved directly！

### Versions

1.8.1+cu11

cc @ezyang @gchanan @zou3519 @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo

**Comment by dzdang** [comment]
@edpzou
I just want to clarify that you mean the output of `torch.fake_quantize_per_tensor_affine(x2, ....)` is wrong? I took a look at the implementation of this function, and I don't see why `fake_quantize_per_tensor_affine` wouldn't work on a non-contiguous tensor. 

Can you please produce `x2`, `x2.stride()` and the output of `torch.fake_quantize_per_tensor_affine`? Thanks

**Comment by aksenventwo** [comment]
I'm constructing for you：

```python
def forward(self, x: Tensor) -> Tensor:
        if self.stride == 1:
            x1, x2 = x.chunk(2, dim=1)

            min_v, max_v = torch._aminmax(x2)
            scale = (max_v - min_v) / 255
            zero_point = torch.round((0 - min_v) / scale).to(torch.int)
            print(x2.is_contiguous(), x.shape, x2.shape, x2.stride())
            x2 = torch.fake_quantize_per_tensor_affine(
                x2, scale.item(), zero_point.item(), 0, 255)
            print(x2.is_contiguous(), x.shape, x2.shape, x2.stride())

            out = torch.cat((x1, self.branch2(x2)), dim=1)
        else:
            out = torch.cat((self.branch1(x), self.branch2(x)), dim=1)

        out = channel_shuffle(out, 2)

        return out
```

outputs:

```
False torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (90944, 784, 28, 1)
True torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (45472, 784, 28, 1)
False torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (90944, 784, 28, 1)
True torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (45472, 784, 28, 1)
False torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (90944, 784, 28, 1)
True torch.Size([2, 116, 28, 28]) torch.Size([2, 58, 28, 28]) (45472, 784, 28, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (45472, 196, 14, 1)
True torch.Size([2, 232, 14, 14]) torch.Size([2, 116, 14, 14]) (22736, 196, 14, 1)
False torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (22736, 49, 7, 1)
True torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (11368, 49, 7, 1)
False torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (22736, 49, 7, 1)
True torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (11368, 49, 7, 1)
False torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (22736, 49, 7, 1)
True torch.Size([2, 464, 7, 7]) torch.Size([2, 232, 7, 7]) (11368, 49, 7, 1)
```

**Comment by dzdang** [comment]
looks like the upgrade solved the issue, will close
